cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,207 @@
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(sriovInfos.Nodes)).ToNot(BeZero(), "no node with SR-IOV NICs found")
+
+		nmstate := nmstatev1.NMState{ObjectMeta: metav1.ObjectMeta{Name: "nmstate"}}
+		err = client.Client.Create(context.Background(), &nmstate)
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(client.Client.Delete, context.Background(), &nmstate)
+	})
+
+	Context("when a SriovNetworkNodePolicy and NodeNetworkConfigurationPolicySpec target the same NIC", func() {
+
+		BeforeEach(func() {
+			namespaces.CleanPods(testNamespace, client.Client)
+		})
+
+		It("VFs should spawn correctly", func() {
+			node, devices, err := findNonPrimarySriovDevicesAndNode(sriovInfos)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices)).ToNot(BeZero())
+
+			By("Using node " + node)
+
+			testDevice := devices[0]
+			By("Using device " + testDevice.Name)
+
+			By("Configuring an IPv4 address on Physical Function via NMState")
+			nodeNetworkConfigPolicy := nmstatev1.NodeNetworkConfigurationPolicy{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-sriov-device-" + testDevice.Name},
+				Spec: nmstateshared.NodeNetworkConfigurationPolicySpec{
+					NodeSelector: map[string]string{
+						"kubernetes.io/hostname": node,
+					},
+					DesiredState: nmstateshared.State{
+						Raw: nmstateshared.RawState(fmt.Sprintf(nmstateDesiredStateTemplateStaticIP, testDevice.Name)),
+					},
+				},
+			}
+			err = client.Client.Create(context.Background(), &nodeNetworkConfigPolicy)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(client.Client.Delete, context.Background(), &nodeNetworkConfigPolicy)
+
+			waitForNMStatePolicyToBeStable(nodeNetworkConfigPolicy.Name)
+
+			By("Configuring VF on device via SriovNetworkNodePolicy")
+			sriovNetworkNodePolicy, err := sriovnetwork.CreateSriovPolicy(sriovclient, "test-nmstate-", namespaces.SRIOVOperator, testDevice.Name, node, 10, "testnmstateresource", "netdevice")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(sriovclient.Delete, context.Background(), sriovNetworkNodePolicy)
+
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/testnmstateresource")]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}).
+				WithPolling(5*time.Second).
+				WithTimeout(10*time.Minute).
+				Should(Equal(int64(10)), "the number of the observed VFs is not that same as the one sriov created")
+		})
+	})
+})
+
+// findNonPrimarySriovDevicesAndNode returns a node name with its SR-IOV devices. The device list does not include
+// the one used for primary networking, if it is an SR-IOV device.
+// The node with the highest number of SR-IOV devices is returned.
+func findNonPrimarySriovDevicesAndNode(n *cluster.EnabledNodes) (string, []*sriovv1.InterfaceExt, error) {
```

Comment:
The name of this is not super clear, and I feel this is doing too many things all together. It finds the node with the highest number of interfaces, and it finds the interfaces for that (excluding the primary).

In general, I'd split this in two steps:
- find the first node with at least two sriov interfaces
- fetch all the interfaces for that node, filtering out the default interface

Also, is there a need to find the one with the highest number of devices or one with one plus the default interface would be enough?

Reply:
> - find the first node with at least two sriov interfaces
There is no need to have two interfaces. If we have one node with one interface that is not used for primary networking, that node/device pair is good.

>Also, is there a need to find the one with the highest number of devices or one with one plus the default interface would be enough?

No special need. The function can exit early as soon as it finds a good node-device pair.
changing

Reply:
sorry I wasn't clear. I meant having two functions, one  that finds the node and the other one that gives you the interface for that node. I think it's clearer and each function would have a clearer responsibility

Reply:
Sorry, I'm not figuring out how this function division should be:
the first function should find a node, but I need a node with at least an SRIOV device that is not used as a primary NIC. Then, the function should return the node name, and another function takes the node name and returns all the non-Primary NIC. 

If this is what you meant, then the first function would be 99% equal to what we have now. 

If not, please provide some code skeleton

Reply:
Changed the name of the function to `findNodeWithNonPrimarySriovNIC`